third_party/terraform/resources/resource_compute_router_nat.go

```diff
@@ -342,19 +348,40 @@ func resourceComputeRouterNatImportState(d *schema.ResourceData, meta interface{
 	return []*schema.ResourceData{d}, nil
 }
 
-func expandSubnetworks(subnetworks []map[string]interface{}) []*computeBeta.RouterNatSubnetworkToNat {
+func expandSubnetworks(subnetworks []interface{}) []*computeBeta.RouterNatSubnetworkToNat {
 	result := make([]*computeBeta.RouterNatSubnetworkToNat, 0, len(subnetworks))
 
 	for _, subnetwork := range subnetworks {
+		snm := subnetwork.(map[string]interface{})
 		subnetworkToNat := computeBeta.RouterNatSubnetworkToNat{
-			Name:                subnetwork["name"].(string),
-			SourceIpRangesToNat: convertStringSet(subnetwork["source_ip_ranges_to_nat"].(*schema.Set)),
+			Name:                snm["name"].(string),
+			SourceIpRangesToNat: convertStringSet(snm["source_ip_ranges_to_nat"].(*schema.Set)),
 		}
-		if v, ok := subnetwork["secondary_ip_range_names"]; ok {
+		if v, ok := snm["secondary_ip_range_names"]; ok {
 			subnetworkToNat.SecondaryIpRangeNames = convertStringSet(v.(*schema.Set))
 		}
 		result = append(result, &subnetworkToNat)
 	}
 
 	return result
 }
+
+func flattenRouterNatSubnetworkToNatBeta(subnetworksToNat []*computeBeta.RouterNatSubnetworkToNat) []map[string]interface{} {
```

Comment:
FYI - I had to re-add this because directly assigning the block didn't work: 

2018/11/30 00:36:07 [DEBUG] Unlocking "router/us-central1/router-nat-test-74df4tnpuu"
2018/11/30 00:36:07 [DEBUG] Unlocked "router/us-central1/router-nat-test-74df4tnpuu"
panic: Invalid address to set: []string{"subnetwork", "0", "SourceIpRangesToNat"}

goroutine 378 [running]:
github.com/terraform-providers/terraform-provider-google-beta/vendor/github.com/hashicorp/terraform/helper/schema.(*ResourceData).Set(0xc0001eeb60, 0x1ebb8ed, 0xa, 0x19adc40, 0xc0005750e0, 0x0, 0x0)
	/usr/local/google/home/shikhman/go/src/github.com/terraform-providers/terraform-provider-google-beta/vendor/github.com/hashicorp/terraform/helper/schema/resource_data.go:191 +0x334
github.com/terraform-providers/terraform-provider-google-beta/google-beta.resourceComputeRouterNatRead(0xc0001eeb60, 0x1aff700, 0xc000280780, 0xf, 0x1ec1847)
	/usr/local/google/home/shikhman/go/src/github.com/terraform-providers/terraform-provider-google-beta/google-beta/resource_compute_router_nat.go:253 +0x8fb